examples/ignite_simple.py

```diff
@@ -0,0 +1,145 @@
+"""
+Optuna example that optimizes convolutional neural networks using PyTorch Ignite.
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+PyTorch Ignite and MNIST. We optimize the neural network architecture as well as the optimizer
+configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
+subset of it.
+
+We have the following two ways to execute this example:
+
+(1) Execute this code directly.
+    $ python ignite_simple.py
+
+
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --direction maximize --storage sqlite:///example.db`
+    $ optuna study optimize ignite_simple.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+from ignite.engine import create_supervised_evaluator
+from ignite.engine import create_supervised_trainer
+from ignite.engine import Events
+from ignite.metrics import Accuracy
+import torch
+from torch import nn
+import torch.nn.functional as F
+from torch.optim import Adam
+from torch.optim import SGD
+from torch.utils.data import DataLoader
+from torch.utils.data import Subset
+from torchvision.datasets.mnist import MNIST
+from torchvision.transforms import Compose
+from torchvision.transforms import Normalize
+from torchvision.transforms import ToTensor
+
+import optuna
+
+
+EPOCHS = 10
+TRAIN_BATCH_SIZE = 64
+VAL_BATCH_SIZE = 1000
+N_TRAIN_EXAMPLES = 3000
+N_VALID_EXAMPLES = 1000
+
+
+class Net(nn.Module):
+    def __init__(self, trial):
+        # We optimize dropout rate in a convolutional neural network.
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        dropout_rate = trial.suggest_uniform('dropout_rate', 0, 1)
+        self.conv2_drop = nn.Dropout2d(p=dropout_rate)
+        self.fc1 = nn.Linear(320, 50)
```

Comment:
(Optional) I suggest tuning the size of features in a FC layer.

```py
fc2_input_dim = trial.suggest_int('fc2_input_dim', 40, 80)
self.fc1 = nn.Linear(320, fc2_input_dim)
self.fc2 = nn.Linear(fc2_input_dim, 10)
```

Reply:
Sounds good!
I'll fix it. Thank you.